client/pwa/src/db.ts

```diff
@@ -57,9 +81,51 @@ export class MDNOfflineDB extends Dexie {
       watched: "url, title, path",
       notifications: "id, title, text, url, created, read, starred",
     });
+    this.version(2).stores({
+      contentStatusHistory: "++id",
+    });
   }
 }
 
 const offlineDb = new MDNOfflineDB();
 
-export { offlineDb };
+async function getContentStatus(): Promise<ContentStatus> {
+  const current = await offlineDb.contentStatusHistory.toCollection().last();
+
+  return (
+    current || {
+      phase: ContentStatusPhase.initial,
+      local: null,
+      remote: null,
+      progress: null,
+      timestamp: new Date(),
+    }
+  );
+}
+
+async function patchContentStatus(
```

Comment:
neat. :)

Reply:
What's possibly less neat is that these helpers are not in a separate file. 🤷